src/mscorlib/shared/System/Buffers/TlsOverPerCoreLockedStacksArrayPool.cs

```diff
@@ -289,6 +410,76 @@ public T[] TryPop()
                 Monitor.Exit(this);
                 return arr;
             }
+
+            public void Trim(uint tickCount, int id, MemoryPressure pressure, int bucketSize)
+            {
+                const uint StackTrimAfterMS = 60 * 1000;                        // Trim after 60 seconds for low/moderate pressure
+                const uint StackHighTrimAfterMS = 10 * 1000;                    // Trim after 10 seconds for high pressure
+                const uint StackRefreshMS = StackTrimAfterMS / 4;               // Time bump after trimming (1/4 trim time)
+                const int StackLowTrimCount = 1;                                // Trim one item when pressure is low
+                const int StackMediumTrimCount = 2;                             // Trim two items when pressure is moderate
+                const int StackHighTrimCount = MaxBuffersPerArraySizePerCore;   // Trim all items when pressure is high
+                const int StackLargeBucket = 16384;                             // If the bucket is larger than this we'll trim an extra when under high pressure
+                const int StackModerateTypeSize = 16;                           // If T is larger than this we'll trim an extra when under high pressure
+                const int StackLargeTypeSize = 32;                              // If T is larger than this we'll trim an extra (additional) when under high pressure
+
+                if (_count == 0)
+                    return;
+
+                lock (this)
+                {
+                    uint trimTicks = pressure == MemoryPressure.High ? StackHighTrimAfterMS : StackTrimAfterMS;
```

Comment:
Nit: this can be moved to before the lock

Reply:
Moving